### ISSUE TITLE
Tests are failing in clean chroot (How to run tests?)

### ISSUE THREAD

**Issue by orhun** [kdheepak/taskwarrior-tui#109]
**Describe the bug**

Hi! Arch Linux maintainer here.
Recently I came across this issue while building the version `0.10.2` of `taskwarrior-tui`:

```
$ cargo test --release --locked -- --skip "test_main"
[...]
running 5 tests
test table::tests::table_invalid_percentages ... ok
test config::tests::test_colors ... ok
test app::tests::test_app ... ok
test app::tests::test_task_style ... FAILED
test app::tests::test_task_tags ... FAILED

failures:

---- app::tests::test_task_style stdout ----
thread 'app::tests::test_task_style' panicked at 'called `Option::unwrap()` on a `None` value', src/app.rs:1747:38

---- app::tests::test_task_tags stdout ----
thread 'app::tests::test_task_tags' panicked at 'called `Option::unwrap()` on a `None` value', src/app.rs:1715:38


failures:
    app::tests::test_task_style
    app::tests::test_task_tags

test result: FAILED. 3 passed; 2 failed; 0 ignored; 0 measured; 0 filtered out; finished in 0.15s

error: test failed, to rerun pass '--bin taskwarrior-tui'
```

Then I realized you're using some kind of [test data](https://github.com/kdheepak/taskwarrior-testdata) for the [test suite](https://github.com/kdheepak/taskwarrior-tui/blob/42bbc091681d07162eb82d120e7655ee6f68185c/.github/workflows/ci.yml#L20). So I disabled the tests for now and wanted to ask: what's the correct way of testing? Is it possible to `cargo test` without test data? Is the error that I'm getting related to the lack of test data?

P.S. I'm not in favor of adding test data to the PKGBUILD. I'd like to solve it the easiest way possible 😊 

Also, the failing `unwrap()` lines are the following:

https://github.com/kdheepak/taskwarrior-tui/blob/bc2759f9a9a18473a0d87cd3b6a32d868bcd35b8/src/app.rs#L1715

https://github.com/kdheepak/taskwarrior-tui/blob/bc2759f9a9a18473a0d87cd3b6a32d868bcd35b8/src/app.rs#L1747

**To Reproduce**

Uncomment the [lines 20-23](https://github.com/archlinux/svntogit-community/blob/d1b17bf43bd541396a1f2bafe6060f92abc21bfd/trunk/PKGBUILD#L20-L23) in [PKGBUILD](https://github.com/archlinux/svntogit-community/blob/d1b17bf43bd541396a1f2bafe6060f92abc21bfd/trunk/PKGBUILD) and `extra-x86_64-build`. 

**Environment (please complete the following information):**

- Operating System:  `Linux`
- Installation:  `Arch Linux`
- taskwarrior-tui version: `0.10.2`

```bash
taskwarrior-tui --version
```
- task version: `2.5.3`

```bash
task --version
```

**Additional context and information**

**Comment by kdheepak** [comment]
Oh shoot! Sorry I should have told you about this! My bad. You'll need to do the following in order to run tests:

```
git clone https://github.com/kdheepak/taskwarrior-testdata
export TASKRC=`pwd`/taskwarrior-testdata/.taskrc
export TASKDATA=`pwd`/taskwarrior-testdata/.task
cargo test
```

See this file for how I'm running tests using github workflows:

https://github.com/kdheepak/taskwarrior-tui/blob/43670fad4d8931e2f15da6726f648656930d5b40/.github/workflows/ci.yml#L20-L43

I'm maintaining the test data as a [separate repo](https://github.com/kdheepak/taskwarrior-testdata), since I expect there to be a lot of churn in that data.

Also, there's another thing I've been meaning to tell you! I've added a man page. The file is located here: https://github.com/kdheepak/taskwarrior-tui/blob/master/docs/taskwarrior-tui.1.md. I use `pandoc` to convert to a `taskwarrior-tui.1` file. 

```
pandoc docs/taskwarrior-tui.1.md --standalone --to=man -o docs/taskwarrior-tui.1
```

I don't think this is very important though, since the instructions are fairly straightforward and are as part of the TUI anyway. If you are interested in shipping the man page as well, you can use that. 

Thank you so much for maintaining the Arch package!

**Comment by kdheepak** [comment]
> P.S. I'm not in favor of adding test data to the PKGBUILD. I'd like to solve it the easiest way possible 😊

If you are not interested in loading the test data just to run tests, then maybe we can run just a subset of tests that I'll always ensure don't need any additional test data.

**Comment by kdheepak** [comment]
Alternatively, now that I've added tests to the CI here, I won't make releases unless the tests pass. So you can remove tests in the PKGBUILD file? Thoughts?

**Comment by kdheepak** [comment]
Also, I can wrap all the tests in a `if` block but then that would be equivalent to not running tests on your end anyway.

**Comment by orhun** [comment]
> If you are not interested in loading the test data just to run tests, then maybe we can run just a subset of tests that I'll always ensure don't need any additional test data.

That doesn't exactly sound like _the easiest way_ so I guess I'll use the test data.

> ```shell
> git clone https://github.com/kdheepak/taskwarrior-testdata
> export TASKRC=`pwd`/taskwarrior-testdata/.taskrc
> export TASKDATA=`pwd`/taskwarrior-testdata/.task
> cargo test
> ```

Yup, it solved it.

> Alternatively, now that I've added tests to the CI here, I won't make releases unless the tests pass. So you can remove tests in the PKGBUILD file? Thoughts?

It's better to test it on both ends I think.

> Also, I can wrap all the tests in a `if` block but then that would be equivalent to not running tests on your end anyway.

There's no need for that at the moment.

> Also, there's another thing I've been meaning to tell you! I've added a man page.
> [...]
> `pandoc docs/taskwarrior-tui.1.md --standalone --to=man -o docs/taskwarrior-tui.1`

Cool!

I enabled the tests and shipped version `0.10.3` which contains the manpage: https://archlinux.org/packages/community/x86_64/taskwarrior-tui/
([diff](https://github.com/archlinux/svntogit-community/commit/023499f2790f09ed50eacf60e02b84c1762793fa#diff-3e341d2d9c67be01819b25b25d5e53ea3cdf3a38d28846cda85a195eb9b7203a))

> Thank you so much for maintaining the Arch package!

Thank you for this nice project! You can close this issue now, everything seems to be resolved :)

**Comment by kdheepak** [comment]
Awesome! Thank you so much!
